### DISCUSSION TITLE
Loading never stop

### DISCUSSION THREAD

**Discussion by luxorwannabe** [stefanzweifel/sidecar-browsershot#75] [Q&A]
Hi, first of all thank you for **sidecar-browsershot.**..

I have successfully installed it and have it connected to S3 Buckets, I found one zip with random name there, also I can see the **sc-lamdabrowsershot-production-browsershot** function listed in Lamba Functions.
But when I try to run it the browser keeps loading never stop (almost 5 minutes) and ends up with the 503 timeout gateway...

Can you help me with this? Thanks

**Comment by stefanzweifel** [comment]
Hi there,

Just for your information: The zip file with the random name and the Lambda function are generated by [sidecar](https://github.com/hammerstonedev/sidecar). This package sits on top of sidecar. This package doesn't influence much on the "Lambda creation" part.

> But when I try to run it the browser keeps loading never stop (almost 5 minutes) and ends up with the 503 timeout gateway...

I need more information to help you here.
What **exactly** are you trying to run in your browser? 
Can you share the code (routes, controller, etc.) you're actually running. 

What options are you forwarding to `BrowsershotLambda`? 
(Do you specify [a timeout on Browsershot](https://spatie.be/docs/browsershot/v2/miscellaneous-options/setting-the-timeout)?)

By default, the Lambda function generated by Sidecar has a timeout of 300 seconds (5 minutes). ([Sidecar Docs](https://hammerstone.dev/sidecar/docs/main/functions/customization#timeout))

You currently can't manipulate this setting for the Browsershot function directly. You have to either manually update the timeout in the AWS dashboard or set the environment variable `SIDECAR_TIMEOUT` to the number of seconds, after which the Lambda should timeout. For example set it to `SIDECAR_TIMEOUT=30` to timeout after 30 seconds.
